### DISCUSSION TITLE
Multiple Achievement/level in one app

### DISCUSSION THREAD

**Discussion by abanghendri** [cjmellor/level-up#88] [Q&A]
Hi, we have a scenario for ecommerce site
**1. Level based on purchase**
user will get 1 point for every $10 purchase on the site, then user will get a Level (e.g: Bronze Member, Gold Member, etc), user could also lost their points for some reason and the level will be downgraded based on their points
the point can be redeemed for product discount (e.g 1 point = $0.1)

**2. Achievement based on interaction**
User will get points based on their interaction on the website (commenting on blog, give product feedback, connecting their social media and mentioning official account, and so on), then user will get badges based on their points achievement (e.g: Master,  Rookie, Rockstar, etc), and user can have multiple badges for multiple interaction, perhaps grouping interaction for example Master will be given to user that achive feedback with more than 10 upvote, and rockstar will be given to user with 100 uploads, and so on.

so, can I achive this with this package?

**Comment by cjmellor** [comment]
From a quick glance at your requirements, it does seem like the package will let you achieve all this.

**Reply by abanghendri** [reply]
so, how?
LevelUp\Experience\Concerns\GiveExperience only give a user point without context, how to know if the point is for purchase based activity and another point is from regular activiry like commenting and so on?
do I need to extend GiveExperience and Level class for each context?

**Reply by cjmellor** [reply]
The package does what it says on the tin. When building your app, just use the package as described in the docs and if something doesn't work how you hoped, fork the repo and expand it.

**Reply by abanghendri** [reply]
nice, thanks for your kind response @cjmellor, I'll try it
